test/e2e/e2e.go

```diff
@@ -2074,6 +2075,129 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
 						externalPort = int32(clusterUDPPort)
 					}
 
+					ginkgo.By("Hitting the external service on " + externalAddress + " and reaching all the endpoints " + protocol)
+					for i := 0; i < maxTries; i++ {
+						epHostname := pokeEndpointHostname(clientContainerName, protocol, externalAddress, externalPort)
+						responses.Insert(epHostname)
+
+						// each endpoint returns its hostname. By doing this, we validate that each ep was reached at least once.
+						if responses.Equal(nodesHostnames) {
+							framework.Logf("Validated external address %s after %d tries", externalAddress, i)
+							valid = true
+							break
+						}
+					}
+					framework.ExpectEqual(valid, true, "Validation failed for external address", externalAddress)
+				}
+			}
+		})
+	})
+	ginkgo.Context("Validating ExternalIP ingress traffic with added node IPs", func() {
+		ginkgo.BeforeEach(func() {
+			endPoints = make([]*v1.Pod, 0)
+			nodesHostnames = sets.NewString()
+
+			var err error
+			nodes, err = e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, 3)
+			framework.ExpectNoError(err)
+
+			if len(nodes.Items) < 3 {
+				framework.Failf(
+					"Test requires >= 3 Ready nodes, but there are only %v nodes",
+					len(nodes.Items))
+			}
+
+			ginkgo.By("Creating the endpoints pod, one for each worker")
+			for _, node := range nodes.Items {
+				// this create a udp / http netexec listener which is able to receive the "hostname"
+				// command. We use this to validate that each endpoint is received at least once
+				args := []string{
+					"netexec",
+					fmt.Sprintf("--http-port=%d", endpointHTTPPort),
+					fmt.Sprintf("--udp-port=%d", endpointUDPPort),
+				}
+				pod, err := createPod(f, node.Name+"-ep", node.Name, f.Namespace.Name, []string{}, endpointsSelector, func(p *v1.Pod) {
+					p.Spec.Containers[0].Args = args
+				})
+				framework.ExpectNoError(err)
+				endPoints = append(endPoints, pod)
+				nodesHostnames.Insert(pod.Name)
+
+				// this is arbitrary and mutuated from k8s network e2e tests. We aim to hit all the endpoints at least once
+				maxTries = len(endPoints)*len(endPoints) + 30
+			}
+
+			ginkgo.By("Creating an external container to send the traffic from")
+			// the client uses the netexec command from the agnhost image, which is able to receive commands for poking other
+			// addresses.
+			createClusterExternalContainer(clientContainerName, agnhostImage, []string{"--network", "kind", "-P"}, []string{"netexec", "--http-port=80"})
```

Comment:
how does is this external container able to reach 172.18.1.x. Wont the kind network give it a 172.18.0.x address? You would need to add a 172.18.1.x ip onto this container too for it to be able to reach those new external ips.

Reply:
No the Docker subnet for nodes is ` "Subnet": "172.18.0.0/16"`, so  172.18.1.x and 172.18.0.x  are both in that subnet

Reply:
And even if it wasn't, a container on the kind network looking for the out of net ip would ARP to ask "who has this address" which the container with the special ip should respond to right?

Reply:
ah i didnt realize it was a /16 i thought it was /24.

"And even if it wasn't, a container on the kind network looking for the "out of net" ip would ARP to ask "who has this address" which the container with the special ip should respond to right?"

No it wouldn't send an ARP for an IP address not on its subnet. It would send it to its default gateway.